pkg/bugreport/ingress.go

```diff
@@ -0,0 +1,138 @@
+				filename := path.Join(
+					c.rootNamespaceDirPath(),
+					namespace.Name,
+					ingressResourceName,
+					ingress.Name,
+					commandsDirName,
+					strings.Join(cmd, "_"),
+				)
+				if err := runCmdAndWriteToFile(cmd, filename); err != nil {
+					return err
+				}
+			}
+			c.completionSuccess("Collected report for ingress %s/%s", ingress.Namespace, ingress.Name)
+		}
+
+		ingressBackendList, err := c.PolicyClient.PolicyV1alpha1().IngressBackends(namespace.Name).List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+
+		for _, ingressBackend := range ingressBackendList.Items {
+			for _, cmd := range ingressReportCommands(namespace.Name, ingressBackend.Name, ingressBackendResourceName) {
+				filename := path.Join(
+					c.rootNamespaceDirPath(),
+					namespace.Name,
+					ingressBackendResourceName,
+					ingressBackend.Name,
+					commandsDirName,
+					strings.Join(cmd, "_"),
+				)
+				if err := runCmdAndWriteToFile(cmd, filename); err != nil {
+					return err
+				}
+			}
+			c.completionSuccess("Collected report for ingress backend %s/%s", ingressBackend.Namespace, ingressBackend.Name)
+		}
+	}
+
+	return nil
+}
+
+func (c *Config) collectIngressControllerReport() error {
+	ingressControllerLabelsSelectors := []string{
+		nginxIngressControllerLabelSelector,
+		contourIngressControllerLabelSelector,
+	}
+	for _, labelSelector := range ingressControllerLabelsSelectors {
+		if err := c.collectIngressControllerReportByLabelSelector(labelSelector); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (c *Config) collectIngressControllerReportByLabelSelector(labelSelector string) error {
+	namespaceList, err := c.KubeClient.CoreV1().Namespaces().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, namespace := range namespaceList.Items {
+		podList, err := c.KubeClient.CoreV1().Pods(namespace.Name).List(
+			context.Background(),
+			metav1.ListOptions{LabelSelector: labelSelector},
+		)
+		if err != nil {
+			return err
+		}
+
+		for _, pod := range podList.Items {
+			cmds := append(
+				ingressControllerReportCommands(pod.Namespace, pod.Name),
+				ingressReportCommands(pod.Namespace, pod.Name, "pods")...,
+			)
+			for _, cmd := range cmds {
+				filename := path.Join(
+					c.rootNamespaceDirPath(),
+					namespace.Name,
+					rootPodDirName,
+					pod.Name,
+					commandsDirName,
+					strings.Join(cmd, "_"),
+				)
+				if err := runCmdAndWriteToFile(cmd, filename); err != nil {
+					return err
+				}
+			}
+			c.completionSuccess("Collected report for ingress controller %s/%s", pod.Namespace, pod.Name)
+		}
+	}
+
+	return nil
+}
+
+func ingressReportCommands(namespace, resourceName, resourceType string) [][]string {
+	return [][]string{
+		{"kubectl", "get", resourceType, "-n", namespace, resourceName, "-o", "yaml"},
+		{"kubectl", "describe", resourceType, "-n", namespace, resourceName},
```

Comment:
I'm surprised we are not using the Kubernetes Go client since it is already available in the repo.